target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming

```diff
@@ -0,0 +1,27 @@
+do_checksumming_disable() {
+	case "$(board_name)" in
+	asus,rp-ac56)
+		local firmware_mtd=$(find_mtd_part firmware)
+		local rootfs_mtd=$(find_mtd_part rootfs)
+		local rootfs_data_mtd=$(find_mtd_part rootfs_data)
+		local rootfs_len=$(grep \"rootfs\" /proc/mtd | awk -F' ' '{print "0x"$2}')
+		local rootfs_data_len=$(grep \"rootfs_data\" /proc/mtd | awk -F' ' '{print "0x"$2}')
+		local offset=$(echo "$rootfs_len $rootfs_data_len 0x40" | awk -F' ' '{printf "%i",$1-$2-$3}')
+		local signature=$(dd if=$rootfs_mtd skip=$offset bs=1 count=4 2>/dev/null | hexdump -v -n 4 -e '1/1 "%02x"')
+		if [ "$signature" = "27051956" ]; then
+			dd conv=notrunc if=$firmware_mtd of=/tmp/header_orig.bin bs=1 count=64 2>/dev/null
+			dd conv=notrunc if=$rootfs_mtd skip=$offset of=/tmp/header_bkp.bin bs=1 count=64 2>/dev/null
```

Comment:
bs=64 count=1 iflag=skip_bytes

Reply:
Done although I'd prefer readability vs performance here

Reply:
I don't really see why it's harder to read now, because indeed readability is important for a code block like this. For the hexdump, I even think it's easier to read now.

Anyway, you need the iflag=skip_bytes, otherwise the offset will be the number of blocks, not bytes.

Reply:
yes, hexdump-based reading of the signature is much better and easier to read. what I meant is that if I use block size 64, it is not clear why block of 64, I have to use an additional flag and command line gets longer. And actually it was a good example: at the end I forgot to add a flag.